### PR TITLE
MAPSJS-2660: Support off-center projection in MapControls.

### PR DESCRIPTION
**Pull request by atomicsulfate**

### CODE REVIEW

File: @here/harp-mapview/test/UtilsTest.ts [hunk 1]
```diff
@@ -209,54 +221,60 @@ describe("MapViewUtils", function () {
...
                 expect(tilt).to.be.closeTo(
-                    tiltLimit,
+                    maxTiltAngle,
                     projection === sphereProjection
                         ? 1e-7 // FIXME: Is this huge error expected?
                         : Number.EPSILON
                 );
             });
             it("limits tilt when orbiting around screen point", function () {
-                for (const startTilt of [0, 20, 45]) {
```
Comment (nzjony) [inline @here/harp-mapview/test/UtilsTest.ts:219]: why do you remove 20, 45?

File: @here/harp-mapview/lib/Utils.ts [hunk 2]
```diff
@@ -245,28 +270,60 @@ export namespace MapViewUtils {
...
+
+    export function orbitAroundScreenPoint(
+        mapView: MapView,
+        offsetXOrOrbitParams: number | OrbitParams,
+        offsetY?: number,
+        deltaAzimuth?: number,
+        deltaTilt?: number,
+        maxTiltAngle?: number
+    ): void {
+        const ppalPoint = CameraUtils.getPrincipalPoint(mapView.camera, cache.vector2[1]);
```
[the remark this answers is not in the document]
Comment (atomicsulfate) [reply @here/harp-mapview/lib/Utils.ts:291]: changed this one to [0] and the one for the orbit center (below) to [1].

File: @here/harp-mapview/lib/Utils.ts [hunk 3]
```diff
@@ -228,15 +229,39 @@ export namespace MapViewUtils {
...
+     */
+    export interface OrbitParams {
+        /**
+         * Delta azimuth in radians.
+         */
+        deltaAzimuth: number;
+        /**
+         * Delta tilt in radians.
+         */
+        deltaTilt: number;
```
[the remark this answers is not in the document]
Comment (atomicsulfate) [reply @here/harp-mapview/lib/Utils.ts:243]: done for both deltaTilt and deltaAzimuth.

File: @here/harp-mapview/test/UtilsTest.ts (hunk 1 above)
Comment (atomicsulfate) [reply @here/harp-mapview/test/UtilsTest.ts:219]: done

**Review by nzjony — APPROVED** [review]

**Pull request merged**